www/gatsby-node.js

```diff
@@ -580,6 +585,24 @@ exports.onCreateNode = ({ node, actions, getNode, reporter }) => {
   // end Community/Creators Pages
 }
 
+exports.onCreatePage = ({ page, actions }) => {
```

Comment:
If pulling in as a YAML node, I don't think this part should be necessary, can just query in the page component

Reply:
Do you suggest that I should write inline filters in `pages/ecosystem.js`

instead of 
```
  query EcosystemQuery(
    $featuredStarters: [String]!
    $featuredPlugins: [String]!
  ) {
    allStartersYaml(
      filter: {
        fields: { starterShowcase: { slug: { in: $featuredStarters } } }
      }
    ) {
      edges {
...
```
do this?

```
  query EcosystemQuery(
    $featuredStarters: [String]!
    $featuredPlugins: [String]!
  ) {
    allStartersYaml(
      filter: {
        fields: { starterShowcase: { slug: { in: [
          "/gatsby-starter-blog/",
          "/gatsby-starter-netlify-cms/",
           "/gatsby-advanced-blog/",
           "/gatsby-advanced-starter/",
           "/gatsby-starter-lumen/"
        ] } } }
      }
    ) {
      edges {
...

That's the only way I know. Sure I can pull `www/src/data/ecosystem/featured-items.yaml` to the `ecosystem.js` but I don not know how to put it as a variable to the `EcosystemQuery` inside `ecosystem.js` file. Could you give me a hint?

Reply:
You could pull the YAML file into the GraphQL layer:

```
{
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `ecosystem`,
        path: `${__dirname}/src/data/ecosystem/`,
      },
    },
```

and then add the query in `gatsby-node.js` graphql query:

```
query {
	allEcosystemYaml {
    edges {
      node {
        starters
        plugins
      }
    }
  }
}
```

Reply:
Re: the page context and `onCreatePage` -- I take it back, you're right. I got mixed up with the way it was being pulled in.